libdevcore/StructuredLogger.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file StructuredLogger.h
+ * @author Lefteris Karapetsas <lefteris@ethdev.com>
+ * @date 2015
+ *
+ * A simple helper class for the structured logging
+ */
+
+#include "StructuredLogger.h"
+
+#include <ctime>
+#include <json/json.h>
+
+using namespace std;
+
+namespace dev
+{
+
+string StructuredLogger::timePointToString(chrono::system_clock::time_point const& _ts) const
+{
+	// not using C++11 std::put_time due to gcc bug
+	// http://stackoverflow.com/questions/14136833/stdput-time-implementation-status-in-gcc
+
+	// TODO: Format it according to Log event Requirements
+	char buffer[64];
+	time_t time = chrono::system_clock::to_time_t(_ts);
+	tm* ptm = localtime(&time);
+	if (strftime(buffer, sizeof(buffer), m_timeFormat.c_str(), ptm))
+		return string(buffer);
+	return "";
+}
+
+void StructuredLogger::outputJson(Json::Value const& _value, std::string const& _name) const
+{
+	Json::Value event;
```

Comment:
Can this be done before creating the detailed event via a helper method, rather than after?
This would eliminate this method. Since we're already instantiating a class for this whole thing, it's just an idea but Json::Value could be wrapped in a class whose destructor is cout << self << endl;. Also note: Look at how other logging is implemented. cout is not thread safe. Said destructor (or the output method) should have a mutex.